lstchain/image/muon/muon_analysis.py

```diff
@@ -87,19 +89,15 @@ def fit_muon(x, y, image, geom, tailcuts):
     image_clean = image * clean_mask
     muonringparam = muonring.fit(x, y, image_clean)
 
-    dist = np.sqrt(np.power(x - muonringparam.ring_center_x, 2)
-                       + np.power(y - muonringparam.ring_center_y, 2))
-    ring_dist = np.abs(dist - muonringparam.ring_radius)
-    muonringparam = muonring.fit(
-            x, y, image_clean * (ring_dist < muonringparam.ring_radius * 0.4)
-        )
-
-    dist = np.sqrt(np.power(x - muonringparam.ring_center_x, 2) +
-                       np.power(y - muonringparam.ring_center_y, 2))
-    ring_dist = np.abs(dist - muonringparam.ring_radius)
+    # Do an iterative fit removing pixels which are beyond 0.4*ring_radius of the ring (along the radial direction):
+    # (Q: is this effective? Seems anyway unnecessary for clean rings which are those we want in the end)
```

Comment:
this was done for the fit to converge

Reply:
> this was done for the fit to converge

Yes, I was just wondering if it is needed for the good rings, and effective for the shower-contaminated ones. I think it will work at least for cases in which there are very few bad pixels, so fine for me to keep it. I modified the comment.